src/dataDisplay/banner.jsx

```diff
@@ -1,43 +1,198 @@
-'use strict';
-
-import React, { Component } from 'react';
-import BannerItem from './bannerItem';
-import ClassNames from 'classnames';
-import Header from '../atoms/header';
-import Icon from '../dataDisplay/icon';
+import {
+    isUndefined,
+    isFunction,
+    forEachRight,
+} from 'lodash';
 import { Portal } from 'react-portal';
+import ClassNames from 'classnames';
 import PropTypes from 'prop-types';
+import React from 'react';
 import ReactDOM from 'react-dom';
+import BannerItem from './bannerItem';
+import Header from '../atoms/header';
+import Icon from './icon';
+import domUtils from '../utils/domUtils';
+import withTheme from '../styles/withTheme';
+
+const propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    id: PropTypes.oneOfType([
+        PropTypes.number,
+        PropTypes.string,
+    ]).isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    level: PropTypes.oneOf([
+        'error',
+        'purple',
+        'secondary',
+        'success',
+        'teal',
+        'warning',
+    ]),
+    levelIcon: PropTypes.string,
+    message: PropTypes.string,
+    onAfterClose: PropTypes.func,
+    onClose: PropTypes.func.isRequired,
+    onOpen: PropTypes.func,
+    style: PropTypes.shape({}),
+    theme: PropTypes.shape({
+        zIndex: PropTypes.shape({
+            banner: PropTypes.number,
+        }),
+    }).isRequired,
+    title: PropTypes.string,
+    topPosition: PropTypes.number,
+    type: PropTypes.oneOf([
+        'alert',
+        'notification',
+    ]),
+};
+
+const defaultProps = {
+    children: undefined,
+    className: undefined,
+    level: undefined,
+    levelIcon: undefined,
+    message: undefined,
+    onAfterClose: undefined,
+    onOpen: undefined,
+    style: undefined,
+    title: undefined,
+    topPosition: undefined,
+    type: undefined,
+};
 
-class Banner extends Component {
+class Banner extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
             isOpen: props.isOpen,
         };
 
-        this._onAnimationComplete = this._onAnimationComplete.bind(this);
-        this._onBeforeClose = this._onBeforeClose.bind(this);
-        this._onOpen = this._onOpen.bind(this);
+        this.onAnimationComplete = this.onAnimationComplete.bind(this);
+        this.onBeforeClose = this.onBeforeClose.bind(this);
+        this.onClose = this.onClose.bind(this);
+        this.onOpen = this.onOpen.bind(this);
     }
 
-    componentDidUpdate(prevProps, prevState) {
-        if (!prevProps.isOpen && this.props.isOpen) {
+    componentDidMount() {
+        const {
+            isOpen,
+        } = this.state;
+
+        if (isOpen) {
+            this.onOpen();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        const {
+            isOpen: prevIsOpn,
```

Comment:
`prevIsOpn` should be `prevIsOpen`.
I kind of like the convention `wasOpen` for the previous one, and keeping `isOpen` (i.e. don't alias it when you destructure it) for the current one, but the `next` and `prev` prefixes works too (and generally, not just for the special case of `isOpen`).